### PR TITLE
CONSOLE-3323: Switch to admin perspective when user navigates directly to a cluster-scoped route and is in the "All Clusters" perspective 

### PR DESCRIPTION
**Pull request by TheRealJon**
When manually navigating to a known cluster-scoped route, the console will automatically switch to the admin perspective so that the active cluster context is clear.

### CODE REVIEW

File: frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx
Comment (jhadvig) [inline frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx:15]: the `frontend` TC are failing due to typescript error:
```
TypeError: src_1.useQueryParams is not a function
```

```suggestion
  const queryParams = useQueryParams()
  const perspective = queryParams.get('perspective');
```

File: frontend/public/components/utils/link.tsx
Comment (jhadvig) [inline frontend/public/components/utils/link.tsx:37]: Not sure why we need namespace prefixes in the cluster scoped ones?

File: frontend/public/components/utils/link.tsx
Comment (jhadvig) [inline frontend/public/components/utils/link.tsx:43]: is this one valid ?

Comment (jhadvig) [reply frontend/public/components/utils/link.tsx:43]: we should be adding `/monitoring/`

File: frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx
Comment (jhadvig) [reply frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx:15]: @TheRealJon so the issue is still in the CI:

Screenshot 2022-11-28 at 17 22 05

**Comment by jhadvig** [comment]
Hmm so the `frontend` test is still failing with the same issue. Wondering if its cause by the fact that we are calling the hook inside a hook, but I doubt that.

**Comment by TheRealJon** [comment]
> 

Fixed.

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

**Comment by TheRealJon** [comment]
/retest
timed out

**Comment by TheRealJon** [comment]
/retest
timeout

**Review by rhamilto — APPROVED** [review]
/lgtm

**Comment by TheRealJon** [comment]
/retest
timed out

**Comment by TheRealJon** [comment]
/retest
known flake

File: frontend/public/components/utils/link.tsx
Comment (yapei) [inline frontend/public/components/utils/link.tsx:43]: `/catalog` route is in dev perspective

Comment (yapei) [reply frontend/public/components/utils/link.tsx:43]: also do we need list all cluster-scope URL here? it looks like some URLs are missing here, for example if I go to following URLs , I will still stay in ACM perspective 
```
/k8s/cluster/core~v1~Namespace
/k8s/cluster/apiextensions.k8s.io~v1~CustomResourceDefinition
/k8s/cluster/config.openshift.io~v1~Infrastructure/cluster 
```
Screenshot 2023-01-05 at 3 11 57 PM

**Comment by TheRealJon** [comment]
/hold

Rethinking how we can solve this issue. I don't think it's safe to assume single/multi cluster context from navigation routes.

File: frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx
Comment (spadgett) [inline frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx:25]: Note that the admin perspective can be disabled in the cluster config.

File: frontend/public/components/utils/link.tsx
Comment (spadgett) [inline frontend/public/components/utils/link.tsx:36]: I worry about having a hard-coded list of paths that we need to maintain. It also means we won't be able to support routes contributed by plugins.

**Review by spadgett — COMMENTED** [review]
@TheRealJon What is the scenario here? You're in the "All Clusters" perspective and someone sends you a link? Or do you have to manually edit the URL?

cc @KevinFCormier

**Comment by KevinFCormier** [comment]
My thought was to switch the current cluster to local-cluster when using All Clusters, so that ACM links out to console pages will be for the correct cluster. But I suppose if you followed a link to an ACM page, the perspective doesn't change automatically, so my approach would leave that gap.

**Comment by TheRealJon** [comment]
We've decided against this approach for handling cluster context for now. We may need to revisit, but for now, I will close this PR.

**Pull request closed without merging**
